python/hsfs/engine/python.py

```diff
@@ -1052,7 +1052,7 @@ def write_training_dataset(
         if (
             arrow_flight_client.is_query_supported(dataset, user_write_options)
             and len(training_dataset.splits) == 0
-            and len(training_dataset.transformation_functions) == 0
+            and len(feature_view_obj.transformation_functions) == 0
```

Comment:
need to check if `feature_view_obj` is not `None`?

Reply:
Added `None` check for `feature_view_obj` object.